packages/salesforcedx-vscode-core/src/telemetry/telemetry.ts

```diff
@@ -25,11 +25,18 @@ export class TelemetryService {
     return TelemetryService.instance;
   }
 
-  public initializeService(context: vscode.ExtensionContext): void {
+  public initializeService(
+    context: vscode.ExtensionContext,
+    machineId: string
+  ): void {
     this.context = context;
-
+    const isDevMode = machineId === 'someValue.machineId';
```

Comment:
@vazexqi after looking at our data and testing all the relevant scenarios it looks like your suggestion of using `someValue.machineId` is good enough.